### PR TITLE
DNSName key retrieval for list-hosted-zones-by-dns-name

### PR DESCRIPTION
**Pull request by tony-dot-sh**
DNSName key retrieval for list-hosted-zones-by-dns-name per https://docs.aws.amazon.com/Route53/latest/APIReference/API_ListHostedZonesByName.html#API_ListHostedZonesByName_RequestSyntax

### CODE REVIEW

File: moto/route53/responses.py [hunk 1]
```diff
@@ -74,9 +75,11 @@ def sort_key(zone):
 
             zones = route53_backend.get_all_hosted_zones()
             zones = sorted(zones, key=sort_key)
+            template = Template(LIST_HOSTED_ZONES_BY_NAME_RESPONSE)
+            return 200, headers, template.render(zones=zones, dnsname=dnsname)
 
-        template = Template(LIST_HOSTED_ZONES_BY_NAME_RESPONSE)
-        return 200, headers, template.render(zones=zones)
+        # template = Template(LIST_HOSTED_ZONES_BY_NAME_RESPONSE)
```
Comment (bblommers) [inline moto/route53/responses.py:81]: Looks like we can keep the return here, as the code is the same in the `if` and `else` sections

File: moto/route53/responses.py [hunk 2]
```diff
@@ -53,15 +53,16 @@ def list_hosted_zones_by_name_response(self, request, full_url, headers):
         dnsname = query_params.get("dnsname")
 
         if dnsname:
-            dnsname = dnsname[
-                0
-            ]  # parse_qs gives us a list, but this parameter doesn't repeat
-            # return all zones with that name (there can be more than one)
+            dnsname = dnsname[0]
+            if dnsname[-1] != ".":
+                dnsname += "."
```
Comment (bblommers) [inline moto/route53/responses.py:58]: Might be me misreading the test, but I can't actually see a test case where we hit this if-statement?

**Review by bblommers — CHANGES_REQUESTED** [review]

File: moto/route53/responses.py (hunk 1 above)
Comment (tony-dot-sh) [reply moto/route53/responses.py]: Good eye, forgot to get rid of those. Done in 2abf91cab05ef7a903fb393018dd8bd0c6c656d4

File: moto/route53/responses.py (hunk 2 above)
Comment (tony-dot-sh) [reply moto/route53/responses.py:58]: Added add'l test in 2abf91cab05ef7a903fb393018dd8bd0c6c656d4

**Review by bblommers — APPROVED** [review]
LGTM. Thanks @tony-dot-sh!

**Comment by bblommers** [comment]
This is now part of moto >= 2.0.3.dev2

**Pull request merged**